Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -34,6 +34,19 @@ open class NavigationMapView: UIView {
      */
     public var showsCongestionForAlternativeRoutes: Bool = false
     
+    /**
+     Controls wheter to show restricted portions of a route line.
+     
+     Restricted areas are drawn using `routeRestrictedAreaColor` which is customizable.
+     */
+    public var showsRestrictedAreasOnRoute: Bool = false {
+        didSet {
+            if let routes = self.routes {
+                show(routes, legIndex: self.currentLegIndex)
```

Comment:
So if we make resume button to switch on/off of `navigationMapView.showsRestrictedAreasOnRoute` during active navigation, we have the following result, because if we call the `navigationMapView.show(_:legIndex)` directly, the previous stored route points and upcoming route point index are cleared. There'll be a 1 second gap of showing the whole route line when `routeLineTracksTraversal` enabled. Until the next `routeProgress` comes in and re-store the upcoming route point index and `fractionTraveled`, could we get the right route line display.
![switch](https://user-images.githubusercontent.com/48976398/144131992-474cc279-054f-45b3-98ca-09de6b46ce20.gif)

But we could make this as a tail work because right now the CarPlay start with the `showsRestrictedAreasOnRoute` enabled or disabled, The gap is almost non. There's not much case that it would be turned on/off during active navigation.

Reply:
Added a [ticket](https://github.com/mapbox/mapbox-navigation-ios/issues/3636) for tracking this.